include/psa/crypto_sizes.h

```diff
@@ -645,6 +645,11 @@
 #undef PSA_SIGNATURE_MAX_SIZE
 #define PSA_SIGNATURE_MAX_SIZE      PSA_BITS_TO_BYTES(PSA_VENDOR_RSA_MAX_KEY_BITS)
 #endif
+#if defined(PSA_WANT_KEY_TYPE_DH_KEY_PAIR_BASIC) && \
```

Comment:
I don't think that's correct: there is not signature algorithm that uses DH keys. (There would be DSA, but we don't and won't support it, and even if we did it would probably have its own key type.

Reply:
I added this piece of code when I encountered failures in `test_suite_psa_crypto_op_fail` of the following type:
```
PSA key_agreement ECDH: incompatible with DH_KEY_PAIR(RFC7919) .... FAILED
  expected_status == psa_raw_key_agreement(alg, key_id, public_key, public_key_length, output, sizeof(output), &length)
  at line 383, suites/test_suite_psa_crypto_op_fail.function
  lhs = 0xffffffffffffff79 = -135
  rhs = 0xffffffffffffff76 = -138
```
As a note `-138` is `PSA_ERROR_BUFFER_TOO_SMALL`. 
The initial check  that fails in `psa_raw_key_agreement` is:
```
if (output_size < expected_length) {
        status = PSA_ERROR_BUFFER_TOO_SMALL;
        goto exit;
    }
```
and this is called in `key_agreement_fail` as follows:
```
    TEST_STATUS(expected_status,
                psa_raw_key_agreement(alg, key_id,
                                      public_key, public_key_length,
                                      output, sizeof(output), &length));
```
where `output` is sized like this:
```
uint8_t output[PSA_SIGNATURE_MAX_SIZE]
```

Reply:
[I forgot to end my comment above...]
So I think that the proper way is to size the `output` buffer with `PSA_RAW_KEY_AGREEMENT_OUTPUT_MAX_SIZE` instead

Reply:
> So I think that the proper way is to size the output buffer with PSA_RAW_KEY_AGREEMENT_OUTPUT_MAX_SIZE instead

That's right. `PSA_SIGNATURE_MAX_SIZE` there is probably copypasta.